packages/neon_framework/lib/src/app.dart

```diff
@@ -236,6 +240,49 @@ class _NeonAppState extends State<NeonApp> with WidgetsBindingObserver, WindowLi
                   darkTheme: appTheme.darkTheme,
                   routerConfig: _routerDelegate,
                 );
+
+                if (activeAccountSnapshot.hasData) {
+                  final account = activeAccountSnapshot.requireData!;
+
+                  return MultiProvider(
+                    providers: [
+                      Provider<Account>.value(
+                        value: account,
+                      ),
+                      Provider<BuiltList<Account>>.value(
+                        // TODO: Might need to listen here.
+                        value: _accountsBloc.accounts.value,
+                      ),
+                      NeonProvider<AccountOptions>.value(
+                        value: _accountsBloc.getOptionsFor(account),
+                      ),
+                      NeonProvider<AppsBloc>.value(
+                        value: _accountsBloc.getAppsBlocFor(account),
+                      ),
+                      NeonProvider<CapabilitiesBloc>.value(
+                        value: _accountsBloc.getCapabilitiesBlocFor(account),
+                      ),
+                      NeonProvider<UserDetailsBloc>.value(
+                        value: _accountsBloc.getUserDetailsBlocFor(account),
+                      ),
+                      NeonProvider<UserStatusBloc>.value(
+                        value: _accountsBloc.getUserStatusBlocFor(account),
+                      ),
+                      NeonProvider<UnifiedSearchBloc>.value(
+                        value: _accountsBloc.getUnifiedSearchBlocFor(account),
+                      ),
+                      NeonProvider<WeatherStatusBloc>.value(
+                        value: _accountsBloc.getWeatherStatusBlocFor(account),
+                      ),
+                      NeonProvider<MaintenanceModeBloc>.value(
+                        value: _accountsBloc.getMaintenanceModeBlocFor(account),
+                      ),
+                    ],
+                    child: app,
+                  );
+                }
```

Comment:
I don't like having to specify this.
Can't we do some more magic 🧙‍♂️ by resolving the required bloc based on it's type in the NeonProvider itself?
https://github.com/nextcloud/neon/blob/6e9cbfbcd586fb4f819b4aee6afc98323af9e0ae/packages/neon_framework/lib/src/utils/provider.dart#L60-L61
This should also revert to the old way where only the AccountsBloc needs to be injected into the context while still being able to resolve other blocs (reverting quite a few changes of this PR)

Reply:
I don't understand what you mean. Sure we could somehow create some matching logic, but it would still have to call a specific function based on the type which would end up with almost the same code and just a switch statement instead?

Reply:
You are right. My idea was to just have a function that dynamically loads the type needed instead of always attaching it to a widget. The current solution will instantiate all blocs at once while my idea was to only create them on the fly when accessed.

  I was not able to find any noticable performance regression. Let's keep it this way for now.

Reply:
Ok now I see your point, but except for UnifiedSearchBloc and WeatherStatusBloc all of the blocs are always used and instantiated anyway.